sql/core/src/test/resources/sql-tests/analyzer-results/subquery/in-subquery/in-with-cte.sql.out

```diff
@@ -198,23 +198,20 @@ WithCTE
          :        :  :     :  +- Filter (cast(t1b#x as int) > 0)
          :        :  :     :     +- SubqueryAlias cte1
          :        :  :     :        +- CTERelationRef xxxx, true, [t1a#x, t1b#x]
-         :        :  :     +- Project [t1b#x AS t1b#x]
-         :        :  :        +- Project [t1b#x]
-         :        :  :           +- Filter (cast(t1b#x as int) > 5)
-         :        :  :              +- SubqueryAlias cte1
-         :        :  :                 +- CTERelationRef xxxx, true, [t1a#x, t1b#x]
-         :        :  +- Project [t1b#x AS t1b#x]
-         :        :     +- Intersect false
-         :        :        :- Project [t1b#x]
-         :        :        :  +- SubqueryAlias cte1
-         :        :        :     +- CTERelationRef xxxx, true, [t1a#x, t1b#x]
-         :        :        +- Project [t1b#x]
-         :        :           +- SubqueryAlias cte1
-         :        :              +- CTERelationRef xxxx, true, [t1a#x, t1b#x]
-         :        +- Project [t1b#x AS t1b#x]
-         :           +- Project [t1b#x]
-         :              +- SubqueryAlias cte1
-         :                 +- CTERelationRef xxxx, true, [t1a#x, t1b#x]
+         :        :  :     +- Project [t1b#x]
+         :        :  :        +- Filter (cast(t1b#x as int) > 5)
+         :        :  :           +- SubqueryAlias cte1
+         :        :  :              +- CTERelationRef xxxx, true, [t1a#x, t1b#x]
```

Comment:
@dtenedor This PR exposes an issue that `DeduplicateRelations` never worked for the CTE ref node, as it's always a new instance and `DeduplicateRelations` can't detect its duplication.

I think the plan change is reasonable as we now deduplicate CTE ref nodes and save one `Project` node.

Reply:
Sounds good. Do you think this test is sufficient to cover the change? Should we try to add more CTE tests with different duplicates? (I am not sure of the answer, just asking the question.)